projects/laji/src/app/shared-modules/datatable/datatable/datatable.component.ts

```diff
@@ -273,4 +309,89 @@ export class DatatableComponent implements AfterViewInit, OnInit, OnDestroy {
       this.logger.info('selected row index failed', e);
     }
   }
+
+  private sortRows(sorts: DatatableSort[] = []) {
+    if (this.sortSub) {
+      this.sortSub.unsubscribe();
+    }
+
+    if (this.clientSideSorting && this._rows) {
+      this.sortLoading = true;
+      this.sortSub = this.sort(sorts, this._rows)
+      .subscribe(() => {
+          this._rows = [...this._rows];
+          this.sortLoading = false;
+          this.changeDetectorRef.markForCheck();
+        }
+      );
+    } else {
+      this.sortLoading = false;
+    }
+  }
+
+  private sort(sorts: DatatableSort[], rows: any[]): Observable<any> {
+    return this.setSortValues(sorts, rows)
+      .pipe(map(() => {
+        if (sorts.length > 0) {
+          this.customSort(sorts, rows);
+        } else {
+          this.defaultSort(rows);
+        }
+      })
+    );
+  }
+
+  private setSortValues(sorts: DatatableSort[], rows: any[]): Observable<any[]> {
+    const obs = sorts.reduce((arr, sort) => {
+      const template = this.sortTemplates[sort.prop];
+      rows.forEach((row) => {
+        if (!this.sortValues[row.preSortIndex]) {
+          this.sortValues[row.preSortIndex] = {};
+        }
+
+        if (this.sortValues[row.preSortIndex][sort.prop] == null) {
+          const rowValue = Util.parseJSONPath(row, sort.prop);
+
+          if (!template || rowValue == null) {
+            this.sortValues[row.preSortIndex][sort.prop] = rowValue;
+          } else {
+            arr.push(
+              this.dtUtil.getVisibleValue(rowValue, row, template)
+                .pipe(tap(val => {
+                  this.sortValues[row.preSortIndex][sort.prop] = val;
+                }))
+            );
+          }
+        }
+      });
+      return arr;
+    }, []);
+
+    return (obs.length > 0 ? forkJoin(obs) : of([]));
+  }
+
+  private customSort(sorts: DatatableSort[], results: any[]) {
+    return results.sort((a: any, b: any) => {
+      for (const sort of sorts) {
+        const dir = sort.dir === 'asc' ? 1 : -1;
+        const aa = this.sortValues[a.preSortIndex]?.[sort.prop] || a[sort.prop];
+        const bb = this.sortValues[b.preSortIndex]?.[sort.prop] || b[sort.prop];
+        const comparison = dir * orderByComparator(aa, bb);
+
+        if (comparison !== 0) {
+          return comparison;
+        }
+      }
+
+      return a.preSortIndex < b.preSortIndex ? -1 : 1;
+    });
+  }
+
+  private defaultSort(results: any[]) {
+    const values = [...results];
+
+    values.forEach((val) => {
+      results[val.preSortIndex] = val;
+    });
+  }
```

Comment:
Make this immutable and make it clear in the signature:

1. Add return type `any[]`
2. Update algorithm to:
``` 
return [...results].map((val) => {
      results[val.preSortIndex] = val;
    });
```